api/db/migrate/20190122160027_move_published_text_from_projects_to_texts.rb

```diff
@@ -0,0 +1,30 @@
+class MovePublishedTextFromProjectsToTexts < ActiveRecord::Migration[5.0]
+  def up
+    add_column :texts, :published, :boolean, default: false, null: false
+
+    execute <<-SQL.squish
+      UPDATE texts
+        SET published = true 
+        WHERE id IN (SELECT published_text_id FROM projects)
+    SQL
+
+    remove_belongs_to :projects, :published_text, type: :uuid
+  end
+
+  def down
+    add_belongs_to :projects, :published_text, type: :uuid
+
+    execute <<-SQL.squish
```

Comment:
Can someone test this?

I first rolled back without the `INNER JOIN` so my data got into a weird state with mismatched texts.  It looked like it worked, but I just want to be sure with valid data.

Reply:
Or maybe @scryptmouse can tell by just looking at it.

Reply:
Tested. Works.